hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractMessageTask.java

```diff
@@ -279,9 +281,12 @@ protected void sendClientMessage(ClientMessage resultClientMessage) {
         if (asyncSocket == null) {
             connection.write(resultClientMessage);
         } else {
-            ClientMessage.Frame frame = resultClientMessage.getStartFrame();
-            //IOBuffer buf = new IOBuffer(resultClientMessage.getBufferLength(), false);
-            IOBuffer buf = responseBufAllocator.allocate(resultClientMessage.getBufferLength());
+            ClientMessage.Frame frame = resultClientMessage.startFrame;
+            Buffer buf = new ThreadLocalBuffer(
+                    (ThreadLocalBufferAllocator) responseBufAllocator,
+                    resultClientMessage.getBufferLength(),
+                    null);
+//            IOBuffer buf = responseBufAllocator.allocate(resultClientMessage.getBufferLength());
```

Comment:
Shouldn't we uncomment this line and remove constructor to not create a new object with every message? Do I miss something?

Reply:
> Do I miss something?
Yes. Temporarily, we are using explicit buffer creation via ctor just for sake of simplicity, as a first step.

Reply:
I don't understand this. What is the point of using an allocator, if we are going to create a new object with every client message. I do understand that we cycle chunks, but we are still creating garbage on the hot path now.

Reply:
Well... Okay, I'd stay with your solution, we always can simplify it